src/query/storages/common/table_meta/src/meta/v4/snapshot.rs

```diff
@@ -87,9 +87,11 @@ pub struct TableSnapshot {
     /// The metadata of the cluster keys.
     pub cluster_key_meta: Option<ClusterKey>,
     pub table_statistics_location: Option<String>,
-    /// The index info locations.
-    /// One table may have multiple indexes, the key is the index name.
-    pub index_info_locations: Option<BTreeMap<String, Location>>,
+
+    // The inverted index information, key is index name,
+    // value is the version and segments that have been indexed,
+    // those segments do not need to be indexed again when refreshing the index.
+    pub indexes: Option<BTreeMap<String, (String, Vec<Location>)>>,
```

Comment:
So we need to drop the table if the old index is added.

Reply:
we don't need to drop the table, if a new index is added or an old index is changed, a new index version will generated, and we need to run refresh to generate new index data.